cmd/fix/feegrant/allowance.go

```diff
@@ -0,0 +1,100 @@
+package feegrant
+
+import (
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/forbole/bdjuno/v2/modules/feegrant"
+	"github.com/forbole/bdjuno/v2/utils"
+
+	"github.com/forbole/juno/v2/cmd/parse"
+	"github.com/forbole/juno/v2/types/config"
+	"github.com/spf13/cobra"
+
+	"github.com/forbole/bdjuno/v2/database"
+
+	feegranttypes "github.com/cosmos/cosmos-sdk/x/feegrant"
+)
+
+// allowanceCmd returns the Cobra command allowing to fix all things related to fee grant allowance
+func allowanceCmd(parseConfig *parse.Config) *cobra.Command {
+	return &cobra.Command{
+		Use:   "allowance",
+		Short: "Fix granted and revoked allowances to the latest height",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			parseCtx, err := parse.GetParsingContext(parseConfig)
+			if err != nil {
+				return err
+			}
+
+			// Get the database
+			db := database.Cast(parseCtx.Database)
+
+			// Build feegrant module
+			feegrantModule := feegrant.NewModule(parseCtx.EncodingConfig.Marshaler, db)
+
+			// Get latest height
+			height, err := parseCtx.Node.LatestHeight()
+			if err != nil {
+				return fmt.Errorf("error while getting chain latest block height: %s", err)
+			}
+
+			startHeight := config.Cfg.Parser.StartHeight
+
+			// Handle messages realated to feegrant allowance of each block from the start height
+			for h := startHeight; h <= height; h++ {
```

Comment:
Instead of parsing all the blocks from the starting height till the latest height, there is a much more efficient way of fixing past fee grant-related transactions: 
1. Get all past transactions that contained a `MsgGrantAllowance` or `MsgRevokeAllowance`
2. Order the transactions based on the ascending height 
3. Parse the transactions messages

You can see something similar [here](https://github.com/desmos-labs/djuno/blob/main/cmd/fix/profiles/cmd.go#L37)

Reply:
Ok! I have updated the code and tested it locally